checker.cpp

```diff
@@ -1,19 +1,50 @@
 #include <assert.h>
 #include <iostream>
+#include <string.h>
 using namespace std;
 
-bool batteryIsOk(float temperature, float soc, float chargeRate) {
-  if(temperature < 0 || temperature > 45) {
-    cout << "Temperature out of range!\n";
-    return false;
-  } else if(soc < 20 || soc > 80) {
-    cout << "State of Charge out of range!\n";
-    return false;
-  } else if(chargeRate > 0.8) {
-    cout << "Charge Rate out of range!\n";
-    return false;
-  }
-  return true;
+#define TEMP_MIN_THRESHOLD 0.00f
+#define TEMP_MAX_THRESHOLD 45.00f
+
+#define SOC_MIN_THRESHOLD 20.00f
+#define SOC_MAX_THRESHOLD 80.00f
+
+#define CHARGE_RATE_MIN_THRESHOLD 0.00f
+#define CHARGE_RATE_MAX_THRESHOLD 0.80f
+
+bool checkValueInRange(float min , float max, const float &valueToCheck ,const std::string &paramName)
+{
+    if(valueToCheck >= min && valueToCheck <=max) return true;
+    else
+    {
+        cout << paramName << " out of range!" <<endl;
+        return false;
+    }
+}
+
+bool checkTemperatureIsOk(const float temperature)
+{
+    return checkValueInRange(TEMP_MIN_THRESHOLD , TEMP_MAX_THRESHOLD,temperature, "Temperature");
+}
+
+bool checkSocIsOk(const float soc)
+{
+    return checkValueInRange(SOC_MIN_THRESHOLD , SOC_MAX_THRESHOLD,soc , "State of Charge");
+}
+
+bool checkChargeRateIsOk(const float chargeRate)
+{
+    return checkValueInRange(CHARGE_RATE_MIN_THRESHOLD , CHARGE_RATE_MAX_THRESHOLD,chargeRate, "Charge Rate");
+}
+
+bool batteryIsOk(const float temperature,const  float soc,const  float chargeRate) {
+  bool temperatureOkIndicator , socOkIndicator , chargeRateOkIndicator = false ;
+
+  temperatureOkIndicator = checkTemperatureIsOk(temperature);
+  socOkIndicator = checkSocIsOk(soc) ;
+  chargeRateOkIndicator = checkChargeRateIsOk(chargeRate);
+
+  return ( temperatureOkIndicator && socOkIndicator && chargeRateOkIndicator);
```

Comment:
Is there any way to prevent this complexity from increasing with the number of parameters?